test/integration/cookbooks/docker-tests/recipes/default.rb

```diff
@@ -0,0 +1,25 @@
+package 'apparmor'
+package 'build-essential'
+
+include_recipe 'docker'
+ENV['CHEF_DRIVER'] = 'docker'
+
+bash 'build and install chef-provisioning-docker' do
+  cwd '/opt/chef-provisioning-docker'
+  code <<-EOS
+    rm chef-provisioning-docker-*.gem
+    /opt/chef/embedded/bin/gem build chef-provisioning-docker.gemspec
+  EOS
+  action :nothing
+end.run_action(:run)
+
+$:.unshift('/opt/chef-provisioning-docker/lib')
+require 'chef/provisioning/docker_driver/version'
+gem_dir = '/opt/chef-provisioning-docker'
+gem_path = File.join(gem_dir, "chef-provisioning-docker-#{Chef::Provisioning::DockerDriver::VERSION}.gem")
+gem_package 'chef-provisioning-docker' do
```

Comment:
ahh right I didn't read all of this and had already forgotten that I ended up using `gem_package` instead of gem install from the batch resource. I cant remember all the details but I had alot of problems with this whole gem install. I cant remember if it was `chef_gem` but even with doing this at compile time, the gem was not available for the below `require` however, if I immediately SSH'd to the box or simply ran the script a second time, it worked.

Finally this "just worked" and I left it but am happy to give chef_gem another shot.